paper/paper.md

```diff
@@ -0,0 +1,204 @@
+
+### Containers
+
+DDC offers two multidimensional containers designed over the C++ 23 multidimensional array `std::mdspan`:
+
+- `Chunk` an owning container, i.e. it manages the lifetime of the underlying memory allocation,
+- `ChunkSpan` is a non-owning container view over existing memory allocation.
+
+### Strongly-typed labelled indices
+
+DDC employs strongly-typed multi-indices to label dimensions and access data. It introduces two types of indices to access the container's data:
+
+- `DiscreteVector` indices:
+  - strongly-typed labelled integers,
+  - provide a multidimensional array access semantics,
+  - always as fast access as raw multidimensional array,
+- `DiscreteElement` indices:
+  - strongly-typed labelled keys or opaque identifiers,
+  - provide an associative access semantics, as keys in a map container,
+  - potentially slower access, depending on the type of set of `DiscreteElement`.
+
+Unlike `DiscreteVector` indices, users cannot directly interpret the internal representation of `DiscreteElement` and must reason about them based solely on their relative position.
+
+### Sets of `DiscreteElement`
+
+The semantics of DDC containers is to associate data to a set of `DiscreteElement` indices. Let us note that the set of all possible `DiscreteElement` has a total order that is typically established once and for all at program initialisation. Thus to be able to construct a DDC container one must provide a multidimensional set of `DiscreteElement` indices, only these indices can be later used to access the container’s data.
+
+The set of `DiscreteElement` is a customization point of the library. It takes the form of a Cartesian product of the different dimensions. DDC predefines the following sets:
+
+- `DiscreteDomain`, a Cartesian product of intervals of `DiscreteElement` in each dimension,
+- `StridedDiscreteDomain`, a Cartesian product of sets of `DiscreteElement` with a fixed step/stride in each dimension,
+- `SparseDiscreteDomain`, a Cartesian product of explicitly ordered lists of `DiscreteElement` in each dimension.
+
+The performance of the container's data access depends on the properties of the set considered. Indeed the set is used to retrieve the position of a given multi-index relatively to the front multi-index. Thus the performance of this operation depends on the information available in the set.
+
+### Slicing
+
+Similar to `std::mdspan`, DDC containers support slicing through the bracket operator that always returns a `ChunkSpan`. The key property is that the resulting slice’s set of `DiscreteElement` is a subset of the original set.
+
+### Multidimensional algorithms
+
+Finally, DDC offers multidimensional algorithms to manipulate the containers and associated `DiscreteElement` indices. The parallel versions are based on Kokkos providing performance portability. Here is a list as of today:
+
+- `parallel_deepcopy`, copies two `ChunkSpan` that are defined on the same set of `DiscreteElement`,
```

Comment:
not sure about "copies two `ChunkSpan`". There are 2 arguments but I wouldn't say that both are copied

Reply:
Done in 00ac5e5f984eb759649dc8837ad7145258c398c5